doc/source/whatsnew/v1.3.0.rst

```diff
@@ -882,6 +882,7 @@ ExtensionArray
 - Fixed bug where :meth:`Series.idxmax`, :meth:`Series.idxmin` and ``argmax/min`` fail when the underlying data is :class:`ExtensionArray` (:issue:`32749`, :issue:`33719`, :issue:`36566`)
 - Fixed a bug where some properties of subclasses of :class:`PandasExtensionDtype` where improperly cached (:issue:`40329`)
 - Bug in :meth:`DataFrame.mask` where masking a :class:`Dataframe` with an :class:`ExtensionArray` dtype raises ``ValueError`` (:issue:`40941`)
+- Bug in :func:`pyarrow_array_to_numpy_and_mask` when converting a pyarrow array who's data buffer size is not a multiple of dtype size (:issue:`40896`)
```

Comment:
```suggestion
- Bug in the conversion from pyarrow to pandas (e.g. for reading Parquet) with nullable dtypes and a pyarrow array whose data buffer size is not a multiple of dtype size (:issue:`40896`)
```

`pyarrow_array_to_numpy_and_mask` is not a public method, so we should try to avoid mentioning it. So a suggested rewording above. 

Also, I would maybe move this to the `I/O` section

Reply:
I changed that as suggested. Also, let me know if I should rebase to get rid of the small commits.

Reply:
> Also, let me know if I should rebase to get rid of the small commits.

No need to do that, we will squash everything when merging (well, github does that for us ;))